### PR TITLE
refactor: small stableswap refactor and docs update

### PR DESCRIPTION
**Pull request by enthusiastmartin**
This PR refactors stableswap pallet little bit - code duplication and consistent naming.

Also updates code docs and comments.

### CODE REVIEW

File: pallets/stableswap/README.md [hunk 1]
```diff
@@ -4,6 +4,18 @@
 
 Curve/stableswap AMM implementation.
 
+### Overview
+
+Curve style AMM at is designed to provide highly efficient and low-slippage trades for stablecoins.
```
Comment (Roznovjak) [inline pallets/stableswap/README.md:9]:
```suggestion
Curve style AMM, designed to provide highly efficient and low-slippage trades for stablecoins.
```

File: pallets/stableswap/src/lib.rs [hunk 2]
```diff
@@ -250,9 +261,6 @@ pub mod pallet {
 		/// Asset is not in the pool.
 		AssetNotInPool,
 
-		/// Asset is already in the pool.
-		AssetInPool,
```
Comment (Roznovjak) [inline pallets/stableswap/src/lib.rs:254]: breaking change

File: math/src/stableswap/math.rs [hunk 3]
```diff
@@ -21,69 +21,89 @@ const PRECISION: u8 = 1;
 /// D - number of iterations to use for Newton's formula to calculate parameter D ( it should be >=1 otherwise it wont converge at all and will always fail
 /// Y - number of iterations to use for Newton's formula to calculate reserve Y ( it should be >=1 otherwise it wont converge at all and will always fail
 pub fn calculate_out_given_in<const D: u8, const Y: u8>(
-	balances: &[AssetReserve],
+	initial_reserves: &[AssetReserve],
```
Comment (mrq1911) [inline math/src/stableswap/math.rs:24]: `initial_reserves` could be bit misleading as that points out to the past time, but you want current state, i would just call it `reserves` or `current_reserves` if you want to be more exact

Comment (mrq1911) [reply math/src/stableswap/math.rs:24]: i see you call it reserves later anyway :) so keep it consistent

**Review by mrq1911 — APPROVED** [review]
fix reserves and LGTM

File: pallets/stableswap/src/lib.rs (hunk 2 above)
Comment (enthusiastmartin) [reply pallets/stableswap/src/lib.rs:254]: it is not used.

**Review by vgantchev — APPROVED** [review]

**Pull request merged**